### DISCUSSION TITLE
How come adding a package as a peer dependency automatically installs it as a dev dependency?

### DISCUSSION THREAD

**Discussion by salieflewis** [pnpm/pnpm#6179] [Q&A]
I know that this is [explicit](https://pnpm.io/cli/add) in the docs, but I'm just confused why this is the default behavior. Aren't there times where you want to have peer dependencies that don't need to be dev deps?

**Comment by zkochan** [comment]
Peer dependencies should always be installed. Unless they are optional peer dependencies. But even optional peer dependencies you probably want to be installed during the development of your package.

**Reply by salieflewis** [reply]
Yeah I understand that part! But when developing a package itself, why is it that peer dependencies automatically become dev dependencies. I'll provide an example.

```
"peerDependencies": {
    "bignumber.js": "^9.1.1",
    "date-fns": "^2.29.3",
    "graphql": "^16.6.0",
    "graphql-request": "^5.0.0",
    "graphql-tag": "^2.12.6",
    "swr": "^1.3.0",
  },
  "devDependencies": {
    "@graphql-codegen/cli": "3.2.1",
    "bignumber.js": "^9.1.1",
    "date-fns": "^2.29.3",
    "graphql": "^16.6.0",
    "graphql-request": "^5.0.0",
    "graphql-tag": "^2.12.6",
    "swr": "^1.3.0",
}
```

Basically, packages like graphql, graphql-request, graphql-tag, don't need to be dev dependencies, but when I add them in a standard fashion or via `--save-peer` they're automatically added as dev dependencies.

**Reply by zkochan** [reply]
You want to add them as prod dependencies?

**Reply by salieflewis** [reply]
I don't believe so. I just don't want them to be dev dependencies when they're not.

**Reply by iobuhov** [reply]
Well, I think you can do whatever you need, just remove peers from your dev dependencies manually and you are done, YOLO, but I wouldn't recommend that. AFAIK, peers also added to dev deps assuming you need toughs during package development or testing. And also when maintainer tells that it's better to keep them, I would think twice before kicking them off from dev deps.

**Reply by salieflewis** [reply]
Not sure what you mean by "toughs during package development or testing". 

Are you suggesting that if when I pull down a package it gets loaded in as a peer and dev dependency, then that is a decision made by the maintainer?

@iobuhov
